src/FadingCounter.h

```diff
@@ -9,32 +9,41 @@
 #ifndef SQUID_SRC_FADINGCOUNTER_H
 #define SQUID_SRC_FADINGCOUNTER_H
 
-#include <vector>
+#include <array>
+#include <chrono>
+
+using namespace std::chrono_literals;
 
 /// Counts events, forgetting old ones. Useful for "3 errors/minute" limits.
 class FadingCounter
 {
 public:
-    FadingCounter();
+    using Horizon = std::chrono::seconds;
 
-    /// 0=remember nothing; -1=forget nothing; new value triggers clear()
-    void configure(double horizonSeconds);
+    /**
+     * Special values:
+     * \li Horizon::zero() - Remember nothing
+     * \li Horizon::max()  - Forget nothing (the default)
+     *
+     * new value triggers clear(), setting the existing value does not.
+     */
+    void configure(const Horizon &);
 
     void clear(); ///< forgets all events
 
-    int count(int howMany); ///< count fresh, return #events remembered
-    int remembered() const { return total; } ///< possibly stale #events
+    uint64_t count(uint64_t howMany); ///< count fresh, return #events remembered
+    auto remembered() const { return total; } ///< possibly stale #events
 
-    /// read-only memory horizon in seconds; older events are forgotten
-    double horizon;
+    /// read-only memory horizon; older events are forgotten
+    Horizon horizon() const { return horizon_; }
 
 private:
-    const int precision; ///< #counting slots, controls measur. accuracy
-    double delta; ///< sub-interval duration = horizon/precision
+    Horizon horizon_ = Horizon::max();
+    double delta = -1; ///< sub-interval duration = horizon/precision
```

Comment:
Please use std::chrono::duration-based type for this horizon-derived cached value. While you adjust this line, please update the data member description to avoid using the now-gone `precision` field.